### DISCUSSION TITLE
Design Ideas

### DISCUSSION THREAD

**Discussion by Lanithane** [target/goalert#2489] [Ideas]
<details>
  <summary>
    Since design topics are much more open-ended than other bodies of work, I wanted to get a discussion started for anyone to share and talk about any design ideas they may have.
  </summary>
  <br />
  <img src="https://user-images.githubusercontent.com/11381794/177395185-91b736db-5fc5-4589-9140-724227e4db27.png" />
</details>

**Comment by Lanithane** [comment]
- [x] goalert icon route to home page
- [ ] autofocus first field in forms
- [ ] move add buttons to cards (leave fabs for mobile)
- [ ] home dashboard
    - services im on call for
    - "my" alerts
    - see graphs by service select
- [ ] splash pages for no alerts/no data
- [ ] private services/permissions by team
- [ ] first time ux
    - [ ] show wizard link in popup
    - [ ] admin first time user setup
    - [ ] clippy popups
- [x] hide alert tab filters within filter menu when on desktop (and tabs are showing)
- [x] disable paginated list checkbox if no items
- [ ] prioritize schedules and rotations that im a member of
    - user icon instead of a heart?
- [ ] every list page, only show what im a part of
    - show as cards, have quick link icons
    - have toggle to view all services, eps, etc instead
- [ ] use regular pointer on d3 charts
- [ ] alert metrics: update interval if date range type change
    - yearly -> month
    - week -> days
    - daily -> hours
    - etc
- [ ] alert metrics table column, column popover should not be inline
- [ ] alert metrics table don’t highlight cells when on non-clickable cells

**Comment by Lanithane** [comment]
- upload/change profile avatar
- show badge of any open alerts on profile avatar in toolbar
- add number of any open alerts to tab title (e.g. "(4) Open Alerts - GoAlert")

**Comment by mastercactapus** [comment]
An idea for the personalized list pages:
![image](https://user-images.githubusercontent.com/595010/203426377-9c2eb37d-6b72-4f12-bd20-5ada91f3a900.png)

**Reply by mastercactapus** [reply]
@AlaricWhitney is this kind of what you were thinking?

**Reply by AlaricWhitney** [reply]
It could be a start.  I was thinking more along the lines a user might want to see metrics, but there's nothing super obvious that I need to goto services first.  What if there was a way to select metrics from the sidebar nav, which brings you to metrics (which doesn't show anything until you select a service from a dropdown.

**Reply by mastercactapus** [reply]
yeah, I see what you're saying. Maybe this layout would be better for the home/personal/default page since it's focused on actions. Maybe the home page should be only cards for services with > 1 alert, and individual cards for single alerts. If you have no active alerts, maybe just show your on-call status.

For "My Services" maybe it would be more helpful to show metrics or something comparing your services, or even if you just have one, an overview of its health, or the last event even.

@Forfold I know you have had some ideas about using cards more effectively, any thoughts?

**Reply by Lanithane** [reply]
this is definitely along the lines of what i was thinking. a list of cards (maybe one column?) with their common actions at the bottom.

the graphs and metrics i think make sense to be cards on a homepage design, where it defaults to your services but you can use a ServiceSelect to customize what graphs are on your home dash.

**Reply by Lanithane** [reply]
i was thinking the home page could be:
- a list of my open alerts (all services, one list, small feed in corner)
- a list of services im on call for (service cards)
- service metrics graphs for services im a member of (or that I'm on-call for, or customized by a ServiceSelect)

and things like that
